state/applicationoffers_test.go

```diff
@@ -500,16 +517,79 @@ func (s *applicationOffersSuite) addOfferConnection(c *gc.C, offerUUID string) {
 		Username:        "admin",
 		SourceModelUUID: testing.ModelTag.Id(),
 	})
+
+	return app
+}
+
+func (s *applicationOffersSuite) TestRemoveOffersSucceedsWithZeroConnections(c *gc.C) {
+	s.createDefaultOffer(c)
+	ao := state.NewApplicationOffers(s.State)
+	err := ao.Remove("hosted-mysql", false)
+	c.Assert(err, jc.ErrorIsNil)
+	err = s.mysql.Destroy()
+	c.Assert(err, jc.ErrorIsNil)
 }
 
-func (s *applicationOffersSuite) TestRemoveOffersWithConnections(c *gc.C) {
+func (s *applicationOffersSuite) TestRemoveOffersFailsWithConnections(c *gc.C) {
 	offer := s.createDefaultOffer(c)
 	s.addOfferConnection(c, offer.OfferUUID)
 	ao := state.NewApplicationOffers(s.State)
 	err := ao.Remove("hosted-mysql", false)
 	c.Assert(err, gc.ErrorMatches, `cannot delete application offer "hosted-mysql": offer has 1 relation`)
 }
 
+func (s *applicationOffersSuite) TestRemoveOffersFailsWhenLateConnectionAppears(c *gc.C) {
```

Comment:
to be consistent with naming elsewhere, perhaps
TestRemoveOffersFailsWithConnectionsRace